to0scheduler/to0serviceimpl/src/main/java/org/sdo/iotplatformsdk/to0scheduler/config/To0ServiceConfiguration.java

```diff
@@ -0,0 +1,197 @@
+import org.springframework.scheduling.annotation.EnableAsync;
+
+/**
+ * Defines the configuration for the this application. Mainly contains the bean
+ * definitions that are used throughout the application.
+ */
+@Configuration
+@EnableAutoConfiguration
+@EnableAsync
+public class To0ServiceConfiguration {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(To0ServiceConfiguration.class);
+
+  // the number of thread in the ThreadPoolTaskExecutor.
+  private final int threadPoolSize = null != To0PropertiesLoader.getProperty("thread.pool.size")
+      ? Integer.parseInt(To0PropertiesLoader.getProperty("thread.pool.size"))
+      : 10;
+  private final int sessionPoolSize = null != To0PropertiesLoader.getProperty("session.pool.size")
+      ? Integer.parseInt(To0PropertiesLoader.getProperty("session.pool.size"))
+      : 10;
+  private final Duration defaultWaitSeconds =
+      null != To0PropertiesLoader.getProperty("org.sdo.to0.ownersign.to0d.ws")
+          ? Duration.parse(To0PropertiesLoader.getProperty("org.sdo.to0.ownersign.to0d.ws"))
+          : Duration.ofHours(1);
+  private final String redirectInfo =
+      To0PropertiesLoader.getProperty("org.sdo.to0.ownersign.to1d.bo");
+
+
+  /**
+   * Registers the controller {@link To0HealthController}.
+   */
+  @Bean
+  protected To0HealthController to0HealthController() {
+    return new To0HealthController();
+  }
+
+  /**
+   * Registers the controller {@link To0Controller}.
+   */
+  @Bean
+  protected To0Controller to0Controller() throws NumberFormatException, Exception {
+    return new To0Controller(to0Scheduler(), defaultWaitSeconds);
+  }
+
+  /**
+   * Register the controller advice {@link To0ControllerAdvice}.
+   */
+  @Bean
+  protected To0ControllerAdvice sdoControllerAdvice() {
+    return new To0ControllerAdvice();
+  }
+
+  /**
+   * Creates and returns a singleton instance of {@link To0Scheduler}. Calls to this method
+   * returns the same instance, always.
+   */
+  @Bean
+  protected To0Scheduler to0Scheduler() throws NumberFormatException, Exception {
+    return new To0Scheduler(executorService(), to0ClientSessionFactory(),
+        new To0SchedulerEventsImpl(restClient()), new To0ProxystoreImpl(restClient()),
+        sessionPoolSize);
+  }
+
+  /**
+   * Creates and returns a singleton instance of {@link To0ClientSession} using
+   * {@link ObjectFactory}. Calls to this method returns the same instance, always.
+   */
+  @Bean
+  protected ObjectFactory<To0ClientSession> to0ClientSessionFactory() throws Exception {
+    return new ObjectFactory<To0ClientSession>() {
+
+      @Override
+      public To0ClientSession getObject() {
+        return new To0ClientSession(signatureServiceFactory(),
+            Paths.get(".").toUri().resolve(redirectInfo).normalize(),
+            HttpClient.newBuilder().sslContext(sslContextFactory().getObject())
+                .connectTimeout(Duration.ofSeconds(30)).build());
```

Comment:
For all hard-coded values, see if you can define constants at a common location. That would increase readability.

Reply:
Fixed by creating a class-level constant.